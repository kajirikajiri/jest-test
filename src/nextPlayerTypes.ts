export type NextPlayer = "X" | "O" | "";
