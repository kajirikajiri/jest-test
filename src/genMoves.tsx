import React from "react";
import { NextPlayer } from "./nextPlayerTypes";

export const genMoves = (
  changeHistory: (i: number) => void,
  history: NextPlayer[][]
): JSX.Element[] => {
  return history.map((_, move: number) => {
    const desc = move ? "Go to move #" + move : "Go to game start";
    return (
      <li key={move}>
        <button
          onClick={() => {
            changeHistory(move);
          }}
        >
          {desc}
        </button>
      </li>
    );
  });
};
