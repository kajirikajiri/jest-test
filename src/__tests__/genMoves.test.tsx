import React from "react";
import { genMoves } from "../genMoves";
import { NextPlayer } from "../nextPlayerTypes";

test("genMoves", () => {
  const setHistoryIndex = (i: number) => {};
  const history: NextPlayer[][] = [[], [], [], [], [], [], [], [], [], []];
  const components = genMoves(setHistoryIndex, history);
  expect(components[0].props.children.props.children).toEqual(
    "Go to game start"
  );
  expect(components[1].props.children.props.children).toEqual("Go to move #1");
  expect(components[9].props.children.props.children).toEqual("Go to move #9");
  const tree = JSON.stringify(components);
  expect(tree).toMatchSnapshot();
});
