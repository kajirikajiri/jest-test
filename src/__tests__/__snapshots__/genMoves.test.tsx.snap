// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`genMoves 1`] = `"[{\\"type\\":\\"li\\",\\"key\\":\\"0\\",\\"ref\\":null,\\"props\\":{\\"children\\":{\\"type\\":\\"button\\",\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"children\\":\\"Go to game start\\"},\\"_owner\\":null,\\"_store\\":{}}},\\"_owner\\":null,\\"_store\\":{}},{\\"type\\":\\"li\\",\\"key\\":\\"1\\",\\"ref\\":null,\\"props\\":{\\"children\\":{\\"type\\":\\"button\\",\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"children\\":\\"Go to move #1\\"},\\"_owner\\":null,\\"_store\\":{}}},\\"_owner\\":null,\\"_store\\":{}},{\\"type\\":\\"li\\",\\"key\\":\\"2\\",\\"ref\\":null,\\"props\\":{\\"children\\":{\\"type\\":\\"button\\",\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"children\\":\\"Go to move #2\\"},\\"_owner\\":null,\\"_store\\":{}}},\\"_owner\\":null,\\"_store\\":{}},{\\"type\\":\\"li\\",\\"key\\":\\"3\\",\\"ref\\":null,\\"props\\":{\\"children\\":{\\"type\\":\\"button\\",\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"children\\":\\"Go to move #3\\"},\\"_owner\\":null,\\"_store\\":{}}},\\"_owner\\":null,\\"_store\\":{}},{\\"type\\":\\"li\\",\\"key\\":\\"4\\",\\"ref\\":null,\\"props\\":{\\"children\\":{\\"type\\":\\"button\\",\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"children\\":\\"Go to move #4\\"},\\"_owner\\":null,\\"_store\\":{}}},\\"_owner\\":null,\\"_store\\":{}},{\\"type\\":\\"li\\",\\"key\\":\\"5\\",\\"ref\\":null,\\"props\\":{\\"children\\":{\\"type\\":\\"button\\",\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"children\\":\\"Go to move #5\\"},\\"_owner\\":null,\\"_store\\":{}}},\\"_owner\\":null,\\"_store\\":{}},{\\"type\\":\\"li\\",\\"key\\":\\"6\\",\\"ref\\":null,\\"props\\":{\\"children\\":{\\"type\\":\\"button\\",\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"children\\":\\"Go to move #6\\"},\\"_owner\\":null,\\"_store\\":{}}},\\"_owner\\":null,\\"_store\\":{}},{\\"type\\":\\"li\\",\\"key\\":\\"7\\",\\"ref\\":null,\\"props\\":{\\"children\\":{\\"type\\":\\"button\\",\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"children\\":\\"Go to move #7\\"},\\"_owner\\":null,\\"_store\\":{}}},\\"_owner\\":null,\\"_store\\":{}},{\\"type\\":\\"li\\",\\"key\\":\\"8\\",\\"ref\\":null,\\"props\\":{\\"children\\":{\\"type\\":\\"button\\",\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"children\\":\\"Go to move #8\\"},\\"_owner\\":null,\\"_store\\":{}}},\\"_owner\\":null,\\"_store\\":{}},{\\"type\\":\\"li\\",\\"key\\":\\"9\\",\\"ref\\":null,\\"props\\":{\\"children\\":{\\"type\\":\\"button\\",\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"children\\":\\"Go to move #9\\"},\\"_owner\\":null,\\"_store\\":{}}},\\"_owner\\":null,\\"_store\\":{}}]"`;
